tests/assertions.py

```diff
@@ -0,0 +1,26 @@
+import operator
+from contextlib import contextmanager
+from typing import Callable, Iterator, TypeVar
+
+T = TypeVar("T")
+
+
+@contextmanager
+def assert_changes(
+    callable: Callable[[], T],
+    before: T,
+    after: T,
+    operator: Callable[[T, T], bool] = operator.eq,
+) -> Iterator[None]:
+    assert operator(callable(), before)
+    yield
+    assert operator(callable(), after)
+
+
+@contextmanager
+def assert_does_not_change(
+    callable: Callable[[], T], value: T, operator: Callable[[T, T], bool] = operator.eq,
+) -> Iterator[None]:
+    assert operator(callable(), value)
+    yield
+    assert operator(callable(), value)
```

Comment:
Using these raises some errors right now because they are invariant on `T`, e.g.

```
tests/utils/streams/mixins.py:192: error: Argument 1 to "assert_does_not_change" has incompatible type "Callable[[], Mapping[Partition, int]]"; expected "Callable[[], Dict[Partition, int]]"
```

I haven't found a fix that I feel comfortable with though.